### PR TITLE
Darek

### PR DESCRIPTION
**Pull request by chefdarek**
This is not the pull you are seeking

### CODE REVIEW

**Comment by chefdarek** [comment]
ce.OrdinalEncoder(),
    SimpleImputer(), 
    RandomForestRegressor(n_estimators=100, random_state=42, n_jobs=-1)
    Features: ['supplier', 'annual_usage', 'min_order_quantity', 'bracket_pricing', 'quantity', 'quote_date_year', 'quote_date_month', 'material_id', 'diameter', 'wall', 'length', 'num_bends', 'bend_radius', 'end_a_1x', 'end_a_2x', 'end_x_1x', 'end_x_2x', 'end_a', 'end_x', 'num_boss', 'num_bracket', 'other']

**Comment by axrd** [comment]
# Module 01 - Log-Linear Regression, Feature Engineering

**Overall**: Nicely done! That's a darn good score. 

**Feedback**: Comment it more. Go back and clean up your notebook. So many ouputs gave Chrome's JavaScript engine a stroke. Reference your assignment notebook with the file name or commit message!

## Load in the Data
- Glob is super useful, huh? 👍
- Great thinking pre-emptively splitting them all into different dataframes 👍  💯 

## Select Features 
- Proper merging 👍 

## Wrangling your Data
- Account for skew / log-transform 👍 
- Handle temporal data 👍 
- Great job making your code functional! 👍 👏 💯 

## Train/Val/Split
- Proper implementation, using train_test_split twice to create a validation set 👌 

## Fit a Model
- Yup. Nicely done. 

## Validation RMSLE (or RMSE)
- Yup. Now **boost** it 😁

**Comment by axrd** [comment]
# Module 02 - Gradient Boosting, More Feature Engineering

*Overall*: You're grokking this Sprint really well. 
*Feedback*: Thank you for the perfect amount of commenting 😄 
## More Feature Engineering
- What else can you engineer? Refer to today's lecture for ideas!

## Implement Gradient Boosting
- Nicely done. Good implementation of `early stopping`, too 👍 

### Reference:
[This is the best explanation of Gradient Boosting I've found so far](https://www.reddit.com/r/explainlikeimfive/comments/86dx3u/eli5_what_is_gradient_boosting_and_how_can_it_be/), in case you want to check it out.

Also, take a look at [Ryan's lecture notebook](https://colab.research.google.com/drive/1LAGRZ5jjO-35hHHTCOtO2VvxwRdwBy-o#scrollTo=zuzgmaBfYiiI) from today for a great reference of a possible solution.

**Comment by chefdarek** [comment]
Yup. I get the Gradient Boosting. Its understanding how categorical features in connecting df's lead to outcomes. I recognized the SQL format of the frames and using the connectors i.d.s in the features is what made me nerd out of the actual nitty gritty. I was focusing on the content of the table. Not seeing the forest for the trees. I got it now, I still suck at feature engineering. I tried for two days to concat them the way he did yesterday.
